game.py

```diff
@@ -0,0 +1,105 @@
+from collections import defaultdict
+
+from etc.utils import Color, read_words, get_random_word, is_valid_word
+from etc.consts import MAX_ATTEMPTS, INDEX_NOT_FOUND
+
+
+class WordleGame:
+    _WORDS = read_words()
+
+    def __init__(self):
+        self._goal_word = ""
+        self.attempts = 0
+        self.word_found = False
+        self.reset()
+
+    def apply_answer(self, guess):
+        if not is_valid_word(guess):
+            raise ValueError("Invalid word", guess)
+        elif self.attempts == MAX_ATTEMPTS:  # Lose
+            return  # Nothing
+        elif guess == self._goal_word:  # Win
+            self.word_found = True
+
+        self.attempts += 1
+
+        result = []
+        chars_to_idxs = self.__gen_chars_indexes_map()
+
+        for i, _ in enumerate(guess):
+            cur_char_color = self.__calc_char_color(guess, i, chars_to_idxs)
```

Comment:
you must use better naming

Reply:
instead of cur_char_color or something else?

Reply:
First, dont afraid of long names. so `current` and not `cur`.
And i think you dont have to stick to `color` here. use the actual meaning of the color.

Reply:
You didnt fixed.